src/django_idom/components.py

```diff
@@ -1,9 +1,119 @@
+from __future__ import annotations
+
 import os
+from inspect import iscoroutinefunction
+from typing import Any, Callable, Dict, Iterable
 
+from channels.db import database_sync_to_async
 from django.contrib.staticfiles.finders import find
-from idom import component, html
+from django.http import HttpRequest
+from django.urls import reverse
+from django.views import View
+from idom import component, hooks, html, utils
+from idom.types import VdomDict
+
+from django_idom.config import IDOM_CACHE, IDOM_VIEW_COMPONENT_IFRAMES
+from django_idom.types import ViewComponentIframe
+
+
+# TODO: Might want to intercept href clicks and form submit events.
+# Form events will probably be accomplished through the upcoming DjangoForm.
+@component
+def view_to_component(
+    view: Callable | View,
+    compatibility: bool = False,
+    transforms: Iterable[Callable[[VdomDict], Any]] = (),
+    strict_parsing: bool = True,
+    request: HttpRequest | None = None,
+    args: Iterable = (),
+    kwargs: Dict | None = None,
+) -> VdomDict | None:
+    """Converts a Django view to an IDOM component.
+
+    Args:
+        view: The view function or class to convert.
+
+    Keyword Args:
+        compatibility: If True, the component will be rendered in an iframe.
+            Strict parsing does not apply to compatibility mode.
+        transforms: A list of functions that transforms the newly generated VDOM.
+            The functions will be called on each VDOM node.
+        strict_parsing: If True, an exception will be generated if the HTML does not
+            perfectly adhere to HTML5.
+        request: Request object to provide to the view.
+        args: The positional arguments to pass to the view.
+        kwargs: The keyword arguments to pass to the view.
+    """
+    kwargs = kwargs or {}
+
+    # Return the view if it's been rendered via the `async_renderer`
+    rendered_view, set_rendered_view = hooks.use_state(None)
+    if rendered_view:
+        return utils.html_to_vdom(
```

Comment:
I'm also realizing that `lake8_idom_hooks` does not show errors for [hook usages after early returns](https://github.com/idom-team/flake8-idom-hooks/issues/15).